{"name": "删除部门里多余的人", "status": "failed", "statusDetails": {"message": "AssertionError: 断言失败 -|- 实际结果:参数错误 || 预期结果: 成功", "trace": "cls = <class 'tools.data_process.DataProcess'>\nresponse = {'errcode': '100002', 'errmsg': '参数错误'}\nexpect_str = '{\"$.errmsg\":\"成功\"}'\n\n    @classmethod\n    def assert_result(cls, response: dict, expect_str: str):\n        \"\"\" 预期结果实际结果断言方法\n        :param response: 实际响应结果\n        :param expect_str: 预期响应内容，从excel中读取\n        return None\n        \"\"\"\n        # 后置sql变量转换\n        expect_str = rep_expr(expect_str, DataProcess.response_dict)\n        expect_dict = convert_json(expect_str)\n        index = 0\n        for k, v in expect_dict.items():\n            # 获取需要断言的实际结果部分\n            actual = extractor(response, k)\n            index += 1\n            logger.info(f'第{index}个断言,实际结果:{actual} | 预期结果:{v} \\n断言结果 {actual == v}')\n            allure_step(f'第{index}个断言',  f'实际结果:{actual} = 预期结果:{v}')\n            try:\n>               assert actual == v\nE               AssertionError\n\ntools/data_process.py:118: AssertionError\n\nDuring handling of the above exception, another exception occurred:\n\ncases = ['case_048', '测试数据清理', '成员，部门相关', '删除部门里多余的人', '/v1/user/delete?access_token=&$.case_001.access_token&', '', ...]\n\n    @pytest.mark.flaky(reruns=0, reruns_delay=1)\n    def test_main(cases):#cases, get_db\n        # 此处的cases入参来自与 conftest.py  文件中 cases函数，与直接使用 @pytest.mark.parametrize\n        # 有着差不多的效果\n        # 发送请求\n        response, expect, sql = BaseRequest.send_request(cases)\n        # 执行sql\n        # DataProcess.handle_sql(sql, get_db)\n        # 断言操作\n>       DataProcess.assert_result(response, expect)\n\ntest/test_api.py:29: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ncls = <class 'tools.data_process.DataProcess'>\nresponse = {'errcode': '100002', 'errmsg': '参数错误'}\nexpect_str = '{\"$.errmsg\":\"成功\"}'\n\n    @classmethod\n    def assert_result(cls, response: dict, expect_str: str):\n        \"\"\" 预期结果实际结果断言方法\n        :param response: 实际响应结果\n        :param expect_str: 预期响应内容，从excel中读取\n        return None\n        \"\"\"\n        # 后置sql变量转换\n        expect_str = rep_expr(expect_str, DataProcess.response_dict)\n        expect_dict = convert_json(expect_str)\n        index = 0\n        for k, v in expect_dict.items():\n            # 获取需要断言的实际结果部分\n            actual = extractor(response, k)\n            index += 1\n            logger.info(f'第{index}个断言,实际结果:{actual} | 预期结果:{v} \\n断言结果 {actual == v}')\n            allure_step(f'第{index}个断言',  f'实际结果:{actual} = 预期结果:{v}')\n            try:\n                assert actual == v\n            except AssertionError:\n>               raise AssertionError(f'断言失败 -|- 实际结果:{actual} || 预期结果: {v}')\nE               AssertionError: 断言失败 -|- 实际结果:参数错误 || 预期结果: 成功\n\ntools/data_process.py:120: AssertionError"}, "steps": [{"name": "请求地址", "status": "passed", "attachments": [{"name": "请求地址", "source": "7efeb0c8-cd68-4d1a-b035-b1086d83b365-attachment.txt", "type": "text/plain"}], "start": 1622542332323, "stop": 1622542332323}, {"name": "请求头", "status": "passed", "attachments": [{"name": "请求头", "source": "c9ef946a-4bb8-4a1f-b5e4-4e4504abcd8c-attachment.txt", "type": "text/plain"}], "start": 1622542332323, "stop": 1622542332324}, {"name": "请求参数", "status": "passed", "attachments": [{"name": "请求参数", "source": "4f807364-9b54-4151-9c1b-318923e75f2a-attachment.txt", "type": "text/plain"}], "start": 1622542332324, "stop": 1622542332325}, {"name": "上传文件", "status": "passed", "attachments": [{"name": "上传文件", "source": "d31a9522-3c40-47f1-84f4-6e8762240ee0-attachment.txt", "type": "text/plain"}], "start": 1622542332325, "stop": 1622542332325}, {"name": "响应耗时(s)", "status": "passed", "attachments": [{"name": "响应耗时(s)", "source": "2318c355-757c-4a55-9402-4dfb271e8298-attachment.txt", "type": "text/plain"}], "start": 1622542332398, "stop": 1622542332399}, {"name": "响应内容", "status": "passed", "attachments": [{"name": "响应内容", "source": "f82d51c5-d81d-4360-94dc-fb5128831eb7-attachment.txt", "type": "text/plain"}], "start": 1622542332399, "stop": 1622542332399}, {"name": "存储实际响应", "status": "passed", "attachments": [{"name": "存储实际响应", "source": "3c063a09-699e-4414-8f9d-d14d04f5193e-attachment.txt", "type": "text/plain"}], "start": 1622542332400, "stop": 1622542332401}, {"name": "第1个断言", "status": "passed", "attachments": [{"name": "第1个断言", "source": "9ce57f48-5f1e-4bdb-8839-ef4ad541a77d-attachment.txt", "type": "text/plain"}], "start": 1622542332402, "stop": 1622542332402}], "parameters": [{"name": "cases", "value": "['case_048', '测试数据清理', '成员，部门相关', '删除部门里多余的人', '/v1/user/delete?access_token=&$.case_001.access_token&', '', 'get', 'params', '', '{\\n    \"userid\":\"&$.case_012.uid&\"\\n}', '', '', '{\"$.errmsg\":\"成功\"}']"}], "start": 1622542332323, "stop": 1622542332402, "uuid": "cca96d14-57f7-4135-a5e4-09e3fef895e4", "historyId": "be525bf87e23a1a4b4594a7e0ffca1f4", "testCaseId": "881516e7ef2e2aa5436f6a3c3e6a5bcb", "fullName": "test.test_api#test_main", "labels": [{"name": "tag", "value": "flaky(reruns=0, reruns_delay=1)"}, {"name": "parentSuite", "value": "test"}, {"name": "suite", "value": "test_api"}, {"name": "host", "value": "liuchengxudeMacBook-Pro.local"}, {"name": "thread", "value": "19614-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "test.test_api"}]}